### PR TITLE
Support advisories with several affected packages with 'pub' a…

### PR DESCRIPTION
**Pull request by szakarias**
…s ecosystem

- Refactored `_extractAdvisoryDetailsForPackage` to support this corner case which we don't expect occures in practice but it is technically allowed.
- Also fixed a small bug in the test setup

### CODE REVIEW

File: test/get/hosted/advisory_test.dart [hunk 1]
```diff
@@ -190,6 +190,36 @@ Future<void> main() async {
     await ctx.run(['get']);
   });
 
+  testWithGolden('more than one affected package with pub as ecosystem',
```
Comment (sigurdm) [inline test/get/hosted/advisory_test.dart:193]:
```suggestion
  testWithGolden('more than one affected package-version with pub as ecosystem',
```
Or would it be better to actually affect two different packages?

File: test/testdata/goldens/get/hosted/advisory_test/more than one affected package with pub as ecosystem.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# GENERATED BY: test/get/hosted/advisory_test.dart
+
+## Section 0
+$ pub get
+Resolving dependencies...
```
Comment (sigurdm) [inline test/testdata/goldens/get/hosted/advisory_test/more than one affected package with pub as ecosystem.txt:5]: Probably has to be regenerated after https://github.com/dart-lang/pub/pull/4175

Comment (szakarias) [reply test/testdata/goldens/get/hosted/advisory_test/more than one affected package with pub as ecosystem.txt]: Yup, done.

**Review by sigurdm — APPROVED** [review]
